crates/router/src/core/blocklist/utils.rs

```diff
@@ -357,3 +365,250 @@ async fn delete_card_bin_blocklist_entry(
+where
+    F: Send + Clone,
+{
+    let db = &state.store;
+    let merchant_id = &merchant_account.merchant_id;
+    let merchant_fingerprint_secret =
+        get_merchant_fingerprint_secret(state, merchant_id.as_str()).await?;
+
+    // Hashed Fingerprint to check whether or not this payment should be blocked.
+    let card_number_fingerprint = payment_data
+        .payment_method_data
+        .as_ref()
+        .and_then(|pm_data| match pm_data {
+            api_models::payments::PaymentMethodData::Card(card) => {
+                crypto::HmacSha512::sign_message(
+                    &crypto::HmacSha512,
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_card_no().as_bytes(),
+                )
+                .attach_printable("error in pm fingerprint creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
+
+    // Hashed Cardbin to check whether or not this payment should be blocked.
+    let card_bin_fingerprint = payment_data
+        .payment_method_data
+        .as_ref()
+        .and_then(|pm_data| match pm_data {
+            api_models::payments::PaymentMethodData::Card(card) => {
+                crypto::HmacSha512::sign_message(
+                    &crypto::HmacSha512,
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_card_isin().as_bytes(),
+                )
+                .attach_printable("error in card bin hash creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
+
+    // Hashed Extended Cardbin to check whether or not this payment should be blocked.
+    let extended_card_bin_fingerprint = payment_data
+        .payment_method_data
+        .as_ref()
+        .and_then(|pm_data| match pm_data {
+            api_models::payments::PaymentMethodData::Card(card) => {
+                crypto::HmacSha512::sign_message(
+                    &crypto::HmacSha512,
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_extended_card_bin().as_bytes(),
+                )
+                .attach_printable("error in extended card bin hash creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
```

Comment:
this block of code seems to be repeated, match is being made on the same field and only one match arm is handled, instead create an iterator and apply this logic